manticore/core/smtlib/expression.py

```diff
@@ -215,26 +162,62 @@ def __bool__(self):
         raise NotImplementedError("__bool__ for Bool")
 
 
-class BoolVariable(Bool, Variable):
-    def __init__(self, name, *args, **kwargs):
-        super().__init__(name, *args, **kwargs)
+class BoolVariable(Bool):
+    __slots__ = ["_name"]
+
+    def __init__(self, name: str, *args, **kwargs):
+        assert " " not in name
+        super().__init__(*args, **kwargs)
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    def __copy__(self, memo):
+        raise ExpressionException("Copying of Variables is not allowed.")
+
+    def __deepcopy__(self, memo):
+        raise ExpressionException("Copying of Variables is not allowed.")
+
+    def __repr__(self):
+        return "<{:s}({:s}) at {:x}>".format(type(self).__name__, self.name, id(self))
 
     @property
     def declaration(self):
         return f"(declare-fun {self.name} () Bool)"
 
 
-class BoolConstant(Bool, Constant):
+class BoolConstant(Bool):
+    __slots__ = ["_value"]
+
     def __init__(self, value: bool, *args, **kwargs):
-        super().__init__(value, *args, **kwargs)
+        self._value = value
+        super().__init__(*args, **kwargs)
 
     def __bool__(self):
         return self.value
 
+    @property
+    def value(self):
+        return self._value
```

Comment:
Ideally, the property should be there only in a debug mode :<

Reply:
Random thought: maybe we should use the private version in internal code? I *believe* (not confirmed) those attributes might be somehow *hot* and we don't want to use a call internally to get their value if possible

Reply:
Previous maintainers agree with you :
https://github.com/trailofbits/manticore/blob/abb191a3a562f30aba42ec4b4b4b4604c283130f/manticore/core/smtlib/visitors.py#L779
We should strive to present the same API to users, but we could probably replace a lot of internal property uses with direct access.